fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -77,15 +77,36 @@ def accumulate_integer_grid(self, index, max_val):
 
 
 class GridSearch(Reduction):
-    """Learner to perform a grid search given a blackbox estimator algorithm. The approach used
+    """Estimator to perform a grid search given a blackbox estimator algorithm. The approach used
     is taken from section 3.4 of `Agarwal et al. (2018) <https://arxiv.org/abs/1803.02453>`_.
 
+
     :param estimator: The underlying estimator to be used. Must provide a
         fit(X, y, sample_weights) method
 
     :param constraints: Object describing the parity constraints. This provides the reweighting
         and relabelling
     :type constraints: fairlearn.moments.Moment
+
+    :param selection_rule: Specifies the procedure for selecting the best model found by the
+        grid search. At the present time, the only valid value is "tradeoff_optimization" which
+        minimises a weighted sum of the error rate and constraint violation.
+    :type selection_rule: str
+
+    :param constraint_weight: When the `selection_rule` is "tradeoff_optimization" this specifies
+        the relative weight put on the constraint violation when selecting the best model.
+        The weight placed on the error rate will be `1-constraint_weight`
```

Comment:
what is 1-constraint_weight?

Reply:
That's supposed to be a mathematical statement - as in
`error_weight = 1 - constraint_weight`